### PR TITLE
New widget: Calendar

### PR DESCRIPTION
**Pull request by snehmehta**
Ticket: #731

### CODE REVIEW

File: lib/widget/calendar.dart [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+  Map<String, Function> methods() {
+    return {
+      'markCall': (value) => _markCell(value),
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'rowHeight': (value) =>
```
Comment (vusters) [inline lib/widget/calendar.dart:50]: This maybe a problem as the row height should expand to accomodate items. Can it just size to its content?

File: lib/widget/calendar.dart [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+            alignment: Utils.getAlignment(data['config']['alignment']),
+            borderRadius:
+                Utils.getBorderRadius(data['config']['borderRadius']));
+      }
+    }
+  }
+}
+
+class CellConfig {
+  Color? color;
```
Comment (vusters) [inline lib/widget/calendar.dart:96]: need textStyle

Comment (vusters) [inline lib/widget/calendar.dart:96]: use `backgroundColor` for consistency.

File: lib/widget/calendar.dart (hunk 1 above)
Comment (snehmehta) [reply lib/widget/calendar.dart:50]: The package we are using has imposed this constrains on the cell with SizedBox.

File: lib/widget/calendar.dart (hunk 2 above)
Comment (snehmehta) [reply lib/widget/calendar.dart:96]: their is labelStyle, I'll update it to textStyle.

File: lib/widget/calendar.dart [hunk 3]
```diff
@@ -80,31 +115,31 @@ class EnsembleCalendar extends StatefulWidget
...
         cell.config = CellConfig(
-            color: Utils.getColor(data['config']['color']),
-            padding: Utils.getInsets(data['config']['padding']),
-            margin: Utils.getInsets(data['config']['margin']),
-            labelStyle: Utils.getTextStyle(data['config']['labelStyle']),
-            alignment: Utils.getAlignment(data['config']['alignment']),
+            backgroundColor: Utils.getColor(data['config']?['backgroundColor']),
+            padding: Utils.getInsets(data['config']?['padding']),
+            margin: Utils.getInsets(data['config']?['margin']),
+            textStyle: Utils.getTextStyle(data['config']?['textStylek']),
```
Comment (vusters) [inline lib/widget/calendar.dart:192]: spelling

**Pull request merged**